src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -1883,7 +1883,12 @@ void InterpreterSelectQuery::initSettings()
 {
     auto & query = getSelectQuery();
     if (query.settings())
+    {
         InterpreterSetQuery(query.settings(), *context).executeForCurrentContext();
+        // Get rid of the settings clause afterwards, so we don't send them to remote.
+        // And newly non-important settings won't break any remote parser.
+        query.setExpression(ASTSelectQuery::Expression::SETTINGS, {});
```

Comment:
Could you explain how non-important settings were able to break a remote parser? They are not important, so they could been just skipped by the remote parser, couldn't they?

Reply:
The unknown non-important setting is caught here https://github.com/ClickHouse/ClickHouse/blob/master/src/Access/SettingsConstraints.cpp#L200

Reply:
If I understood correctly, the issue that you're going to fix here is that you have a `SELECT` query from a distributed table, and the `SELECT` query contains the `SETTINGS` clause, and you don't want ClickHouse to pass the `SETTINGS` clause to the remote nodes, right?
If so it seems it's would a lot better to modify the [rewriteSelectQuery](https://github.com/ClickHouse/ClickHouse/blob/3e576a29c9289df12f5df4c917bdfe47cbf1df59/src/Storages/StorageDistributed.cpp#L97) function instead.

Reply:
@vitlibar Exactly. And that's what @alexey-milovidov suggested too. I leaved this PR to see if there are corner cases that can be caught by CI, as view and materialized view might be broken. However it seems there isn't. I'll update as adviced.